x-pack/plugins/maps/common/descriptor_types/source_descriptor_types.ts

```diff
@@ -60,7 +60,13 @@ export type FieldedAggDescriptor = AbstractAggDescriptor & {
   field?: string;
 };
 
-export type AggDescriptor = CountAggDescriptor | FieldedAggDescriptor;
+export type PercentileAggDescriptor = AbstractAggDescriptor & {
+  type: AGG_TYPE.PERCENTILE;
+  field?: string;
+  percentile?: number;
```

Comment:
Should percentile be optional? I think it should be required since you are providing a default value when the agg type is switched to percentile so its always populated.

Reply:
discussed offline. Making `percentile` required will break all our use of `AggDescriptor`, because the compiler will require explicit type-checking to handle all the places where it's not explicitly defined. `field` is optional for a similar reason.

Changing this would require a refactor out-of-scope of this PR. (might want to re-evaluate the use of union-types too in this situation).